### ISSUE TITLE
Focusing the Country Input or State Input doesn't read the current value

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#2040]
**To Reproduce**
1. With a screen reader navigate the _Checkout_ block and focus the Country Input.
2. Notice there is no way to know what's the selected value, since its not read by the screen reader.

**Expected behavior**
The screen reader should read what value is currently selected, so the user knows if it's correct or needs to be changed.

**Notes**
Before fixing this, we should find out if the issue is in our `CountryInput` and `StateInput` components, in the `CustomSelectControl` Gutenberg component or in the `downshift` library. Ideally, we would upstream the fix as much as possible.

**Comment by Aljullu** [comment]
I opened an issue in Gutenberg's repo: https://github.com/WordPress/gutenberg/issues/21459 and a PR: https://github.com/WordPress/gutenberg/pull/21461. In case it doesn't get merged, we can make the same changes in our `Select` component, but wanted to try it first upstream.

**Comment by Aljullu** [comment]
I will proceed closing this issue (and remove it from the 2.7.0 milestone cc @nerrad). In my last comment I linked the issue and the PR in Gutenberg. It can get broader a11y testing there, so I would avoid any downstream implementation.

Feel free to reopen if that doesn't sound good. :slightly_smiling_face:
